packages/db/src/translate/executions.ts

```diff
@@ -3,10 +3,68 @@ import Knex from 'knex';
 import { RecordDoesNotExist } from '@cumulus/errors';
 import { ExecutionRecord } from '@cumulus/types/api/executions';
 import Logger from '@cumulus/logger';
-import { PostgresExecution } from '../types/execution';
+import { PostgresExecution, PostgresExecutionRecord } from '../types/execution';
 import { ExecutionPgModel } from '../models/execution';
 import { CollectionPgModel } from '../models/collection';
 import { AsyncOperationPgModel } from '../models/async_operation';
+const { removeNilProperties } = require('@cumulus/common/util');
+
+export const translatePostgresExecutionToApiExecution = async (
+  executionRecord: PostgresExecutionRecord,
+  knex: Knex,
+  collectionPgModel = new CollectionPgModel(),
+  asyncOperationPgModel = new AsyncOperationPgModel(),
+  executionPgModel = new ExecutionPgModel()
+): Promise<ExecutionRecord> => {
+  let parentArn: string | undefined;
+  let collectionId: string | undefined;
+  let asyncOperationId: string | undefined;
+
+  if (executionRecord.collection_cumulus_id) {
+    const collection = await collectionPgModel.get(knex, {
+      cumulus_id: executionRecord.collection_cumulus_id,
+    });
+    collectionId = `${collection.name}___${collection.version}`;
```

Comment:
Same comments as above. We could use a helper here if it's not a pain to import. `constructCollectionId`

Reply:
f135216